pandas/tests/frame/methods/test_combine_first.py

```diff
@@ -492,3 +492,23 @@ def test_combine_preserve_dtypes():
     )
     combined = df1.combine_first(df2)
     tm.assert_frame_equal(combined, expected)
+
+
+def test_combine_first_duplicates_rows_for_nan_index_values():
+    # GH39881
+    df1 = DataFrame({"a": [1, 2, 3], "b": [np.nan, 5, 6], "x": [9, 10, 11]})
+    df1 = df1.set_index(["a", "b"])
```

Comment:
Please define index explicitly when creating the dataframe. Same in the following lines

otherwise lgtm

Reply:
done.